### PR TITLE
Add timeout var for git clone (local recipe)

### PR DESCRIPTION
**Pull request by 42antoine**
When updating project with multiple submodules, git clone timeout as "Nothing happened since 60 seconds" but currently updating submodule.

### CODE REVIEW

File: recipes/local.php [hunk 1]
```diff
@@ -94,6 +94,8 @@
     $branch = env('branch');
     $gitCache = env('local_git_cache');
     $depth = $gitCache ? '' : '--depth 1';
+    $gitCloneTimeout = env('local_git_clone_timeout');
+    $gitCloneTimeout = $gitCloneTimeout ? $gitCloneTimeout : 60;
```
Comment (johnny-bit) [inline recipes/local.php]: You should look at how it's done in rsync recipe. https://github.com/deployphp/recipes/blob/4f21808a3278855ed1e3afdebbf1d48afeb831f2/recipes/rsync.php#L21

**Comment by johnny-bit** [comment]
I'm ok with this in sense of timeout, but would look better if used style recommended in other recipes - single configuration point, whit example in rsync recipe :)

**Comment by 42antoine** [comment]
Okidoki, something like this :

```
set('local', [
    'git' => [
        'timeout' => 60,
    ]
]);

$config = get('local');
$config['git']['timeout']
```

**Comment by johnny-bit** [comment]
hmm...

set('local', [
 'timeout' => 60,
]);

would be enough ;) then you could append that to all methods which can and
should take extra time (not only cloning) and viola.

Plus enhance Your PR with documentation change and i'll have no more
complaints :)

2016-03-07 11:17 GMT+01:00 Antoine <notifications@github.com>:

> Okidoki, something like this :
>
> set('local', [
>     'git' => [
>         'timeout' => 60,
>     ]
> ]);
>
> $config = get('local');
> $config['git']['timeout']
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/deployphp/recipes/pull/41#issuecomment-193193585>.
>



-- 
Pozdrawiam,
Hubert Kowalski

**Comment by 42antoine** [comment]
Taadaa! cf20269

File: docs/local.md [hunk 2]
```diff
@@ -12,7 +12,23 @@ require 'vendor/deployphp/recipes/recipes/local.php';
 
 ### Configuration options
 
-The `local.php` recipe doesn't expose any configuration options. Instead uses same options as standard `common.php` recipe.
+The `local.php` uses same options as standard `common.php` recipe and few specific options.
+
+- **local**: Accepts an array with following local options (all are optional and defaults are ok):
```
Comment (mbrodala) [inline docs/local.md]: Please use `... with the following ...`

File: docs/local.md [hunk 3]
```diff
@@ -12,7 +12,23 @@ require 'vendor/deployphp/recipes/recipes/local.php';
 
 ### Configuration options
 
-The `local.php` recipe doesn't expose any configuration options. Instead uses same options as standard `common.php` recipe.
+The `local.php` uses same options as standard `common.php` recipe and few specific options.
```
Comment (mbrodala) [inline docs/local.md]: Please use `... uses the same options as the standard ... and a few ...`

File: docs/local.md [hunk 4]
```diff
@@ -12,7 +12,23 @@ require 'vendor/deployphp/recipes/recipes/local.php';
...
+
+- **local**: Accepts an array with following local options (all are optional and defaults are ok):
+
+- *timeout*: accepts an *int* defining timeout for local command to run locally.
```
Comment (mbrodala) [inline docs/local.md]: Please use `... defining the timeout for the local command.` (Skip the rest. ;-) )

**Comment by 42antoine** [comment]
Hi! I have updated the branch with last master recipes commit and i have changed the doc. Hope this could be integrated :D

**Comment by elfet** [comment]
Please, rebase your PR.

**Pull request closed without merging**